pickhardtpayments/Attempt.py

```diff
@@ -0,0 +1,134 @@
+from enum import Enum
+
+from pickhardtpayments import Channel
+
+
+class SettlementStatus(Enum):
+    PLANNED = 1
+    INFLIGHT = 2
+    ARRIVED = 3
+    FAILED = 4
+    SETTLED = 5
+
+
+class Attempt:
+    """
+    An Attempt describes a path (a set of channels) of an amount from sender to receiver.
+
+    When sending an amount of sats from sender to receiver, a payment is usually split up and sent across
+    several paths, to increase the probability of being successfully delivered. Each of this path is referred to as an
+    Attempt.
+    An Attempt consists of a list of Channels (class:Channel) and the amount in sats to be sent through this path.
+
+    :param path: a list of UncertaintyChannel objects from sender to receiver
+    :type path: list[UncertaintyChannel]
+    :param amount: the amount to be transferred from source to destination
+    :type amount: int
+    """
+
+    def __init__(self, path: list[Channel], amount: int = 0):
+        """Constructor method
+        """
+        self._paths = None
+        self._routing_fee = -1
+        self._probability = -1
+        if not isinstance(path, list):
+            raise ValueError("path needs to be a collection of Channels")
+        for channel in path:
+            if not isinstance(channel, Channel):
+                raise ValueError("path needs to be a collection of Channels")
+            self._path = path
+        self._status = SettlementStatus.PLANNED
+        self._amount = amount
+
+    def __str__(self):
+        return "Path with {} channels to deliver {} sats and status {}.".format(len(self._path),
```

Comment:
Looking how I depict attempts in the evaluation we might want to change that string to include stuff like `probability`, `fees`, `ppm`. We could get inspiration by the `print`output of the simulation